cmake/vars_utils.cmake

```
@@ -26,12 +26,20 @@ get_filename_component(TBB_VARS_TEMPLATE_NAME ${PROJECT_SOURCE_DIR}/integration/
 string(REPLACE ".in" "" TBB_VARS_NAME ${TBB_VARS_TEMPLATE_NAME})
 
 macro(tbb_gen_vars target)
+    if (NOT TBB_BUILD_APPLE_FRAMEWORKS)
+        set(BIN_PATH $<TARGET_FILE_DIR:${target}>)
+    else()
+        # For Apple frameworks, the binaries are placed in a framework bundle. 
+        # When using an Apple framework, you refer to the bundle, not the binary inside, so we take the bundle's path and go up one level.
+        # This path will then be used to generate the vars file, and the contents of the vars file will use the bundle's parent directory.
```

Following approach as @aepanchi suggested in previous comments.
```suggestion
        # For Apple* frameworks, the binaries are placed in a framework bundle. 
        # When using an Apple* framework, you refer to the bundle, not the binary inside, so we take the bundle's path and go up one level.
        # This path will then be used to generate the vars file, and the contents of the vars file will use the bundle's parent directory.
```